From 167275763b7eb418b083b2eeb489108047b5d81b Mon Sep 17 00:00:00 2001
From: Laurent Carlier <lordheavym@gmail.com>
Date: Tue, 11 Jul 2017 13:42:59 +0200
Subject: [PATCH] pkgconfig: avoid appending slash at Cflags

otherwise it can break some compilations, see https://bugs.archlinux.org/task/54763

(cherry picked from commit 0283a2fb41fb4c25be1d0078bb40ae761d47462a)
---
 mesonbuild/modules/pkgconfig.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/mesonbuild/modules/pkgconfig.py b/mesonbuild/modules/pkgconfig.py
index 09c615ab..c044d91e 100644
--- a/mesonbuild/modules/pkgconfig.py
+++ b/mesonbuild/modules/pkgconfig.py
@@ -96,10 +96,11 @@ class PkgConfigModule(ExtensionModule):
                 ofile.write('Libs.private: {}\n'.format(' '.join(generate_libs_flags(priv_libs))))
             ofile.write('Cflags:')
             for h in subdirs:
-                if h == '.':
-                    h = ''
                 ofile.write(' ')
-                ofile.write(os.path.join('-I${includedir}', h))
+                if h == '.':
+                    ofile.write('-I${includedir}')
+                else:
+                    ofile.write(os.path.join('-I${includedir}', h))
             ofile.write('\n')
 
     def process_libs(self, libs):
-- 
2.13.2

